src/components/create-guild/Requirements/components/ConfirmationAlert.tsx

```diff
@@ -43,7 +43,7 @@ const ConfirmationAlert = ({
               data-test="delete-requirement-button"
               colorScheme="red"
               ml={3}
-              onClick={onConfirm || onClose}
+              onClick={!!onConfirm ? () => onConfirm() : () => onClose()}
```

Comment:
Yup, it's actually needed, because if we just pass the function here like this: `onClick={onConfirm}`, the click event object is automatically passed down to that function as a param, and we don't want that to happen